### ISSUE TITLE
Sprinkler Tracking

### ISSUE THREAD

**Issue by cappa88** [jeroenterheerdt/HAsmartirrigation#458]
I would like to propose an idea. 

Would it be possible to include the entity of the respective sprinklers for each zone? This way, the integration could track the seconds each zone is irrigated and update the bucket accordingly. This could avoid the need to reset the bucket and also account for any manual activations of individual zones.

**Comment by jeroenterheerdt** [comment]
that's a great idea! Very interesting, let me see if I understand what you mean:
1) add optional entity for each zone
2) if entity is provided, automatically update bucket to reflect changes based on throughput * activation time. this will cover then for automatic irrigation as part of the normal behavior as well as manual activations of the entity.

did I get that right?

**Comment by cappa88** [comment]
Yes, that's exactly what I mean! By adding an optional entity for each zone and updating the bucket based on throughput and activation duration, the integration would become more accurate and efficient. The bucket would be adjusted according to the calculation of millimeters of water replenished by the irrigation. This would account for both automatic and manual irrigation, making the system more reliable.Thank you for considering this idea. As I'm not familiar with the development of integrations, I wasn't sure if this was feasible

**Comment by dadino** [comment]
This would solve my problems with an unrealiable water resource. My only source is a tank, which could be empty, meaning the sprinkler is activated, power is given to the pump, but the pump's floater is not high enough to allow the pump to run. 

I have an entity in Home Assistant that checks the power consumption of the pump and the switch for the sprinkler, if both are valid, then I consider the sprinkler to be actually working.

It's a binary sensor, so if you implement this change (which would be a very good addition), don't limit it to switches or input_boolean, please.

**Comment by jeroenterheerdt** [comment]
@dadino sounds like you'd need the ability to specify multiple entities or would you be able to wrap them into one?

**Comment by russdill** [comment]
Related to this, I have sensors that give flow for each zone and also accumulate volume for each zone. It would eliminate a lot of automation work if these could just automatically feed in to HASI. Would that be a separate issue?

**Comment by altmenorg** [comment]
This is a great proposal and the discussion above captured it well. It is on the roadmap and we plan to integrate it: an optional per-zone entity that closes the loop on the bucket, so actual irrigation (automatic and manual) credits the bucket without a manual reset.

The design we are aiming for supports two kinds of input:
- a generic on/off entity (any binary state, not only switches, per @dadino's pump and tank case) where the bucket is credited from throughput x run time, and
- a flow or volume sensor per zone (per @russdill) where the bucket is credited from the measured volume directly. That also covers #626, which is the same feature family.

A detail we will handle carefully: avoiding double counting against the existing irrigation event and bucket reset, so when tracking is enabled it becomes the single source of truth for irrigation credit.

Keeping this open as a planned feature. Thanks @cappa88 for the idea, and @dadino and @russdill for the unreliable-source and flow-sensor cases.

**Comment by altmenorg** [comment]
Implemented on the `dev` branch as **observed watering** (closed-loop bucket).

Link a valve or switch to each zone and the bucket is credited automatically from the real run time and the zone throughput, so there is no manual reset. Manual taps and automations are counted too. Smart Irrigation can optionally drive the valve itself (direct valve control, with open-confirmation, reboot resilience and a per-zone problem event).

This is on `dev` (PR #744) and not in a published release yet. It will ship in the next beta. Docs: `docs/configuration-closed-loop.md`.
